src/coiio/typedesc.cpp

```diff
@@ -0,0 +1,27 @@
+#include <OpenImageIO/typedesc.h>
+
+#include "typedesc.h"
+#include "util.hpp"
+
+// Sanity check that our types are equivalent
+static_assert(sizeof(TypeDesc) == sizeof(OIIO::TypeDesc));
+static_assert(alignof(TypeDesc) == alignof(OIIO::TypeDesc));
+static_assert(offsetof(TypeDesc, basetype)
+              == offsetof(OIIO::TypeDesc, basetype));
```

Comment:
I like the use of static_assert here to ensure that the C and C++ structs stay exactly equivalent.

Reply:
OH, these aren't legal C++11 (only 14) because they lack an explanation string.
Consider using the `OIIO_STATIC_ASSERT` macro in dassert.h, which takes care of this for you.